### PR TITLE
use lzma, not gzip, for cpio archives.

### PR DESCRIPTION
**Pull request by rminnich**
it's about a 3x improvement.

Signed-off-by: Ronald G Minnich <rminnich@gmail.com>

### CODE REVIEW

File: cmd/tmpfs/tmpfs.go [hunk 1]
```diff
@@ -321,7 +321,7 @@ func readImage(n string) (*tmpfs.Archive, error) {
 	}()
 	r := io.Reader(f)
 
-	if arch, err := gzip.NewReader(f); err == nil {
+	if arch, err := lzma.NewReader(f); err == nil {
```
Comment (gmacd) [inline cmd/tmpfs/tmpfs.go:324]: Happy to support lzma given the massive improvement, but should also change the references to "gz" and "tgz" below.

Comment (sevki) [reply cmd/tmpfs/tmpfs.go:324]: two questions, is lzma middle-out? and what's it's Weissman score? :smiley:

Comment (sevki) [reply cmd/tmpfs/tmpfs.go:324]: all joking aside 3x improvement on a single line change is pretty rad :)

Comment (rminnich) [reply cmd/tmpfs/tmpfs.go:324]: no you leave those in, they are for your tar unpacker, which is in a different file. They should stay in.

Comment (rminnich) [reply cmd/tmpfs/tmpfs.go:324]: This code needs a real rework but it works, those names need to stay until you rewrite ReadImageTar()

Comment (rminnich) [reply cmd/tmpfs/tmpfs.go:324]: in the best world, ReadImageTar would take an io.Reader, and we would separate the decompression from the archive steps. But that's for later.

**Review by gmacd — APPROVED** [review]

**Pull request merged**